pkg/telemetry/closeable_dogstatsd_sink.go

```diff
@@ -0,0 +1,35 @@
+package telemetry
+
+import (
+	"reflect"
+	"unsafe"
+
+	"github.com/DataDog/datadog-go/statsd"
+	"github.com/armon/go-metrics/datadog"
+)
+
+type CloseableDogStatsdSink struct {
+	*datadog.DogStatsdSink
+}
+
+var _ CloseableMetricSink = (*CloseableDogStatsdSink)(nil)
+
+func NewCloseableDogStatsdSink(addr string, hostName string) (*CloseableDogStatsdSink, error) {
+	dogStatsdSink, err := datadog.NewDogStatsdSink(addr, hostName)
+	if err != nil {
+		return nil, err
+	}
+
+	sink := &CloseableDogStatsdSink{
+		DogStatsdSink: dogStatsdSink,
+	}
+	return sink, nil
+}
+
+func (s *CloseableDogStatsdSink) Close() {
+	structType := reflect.ValueOf(s.DogStatsdSink).Elem()
+	privateField := structType.FieldByName("client")
+	fieldValue := reflect.NewAt(privateField.Type(), unsafe.Pointer(privateField.UnsafeAddr())).Elem()
+	statsdClient := fieldValue.Interface().(*statsd.Client)
+	_ = statsdClient.Close()
+}
```

Comment:
I'd love to say I figured this out on my own, but no:

https://stackoverflow.com/questions/42664837/how-to-access-unexported-struct-fields/43918797#43918797